libs/expo/betterangels/src/lib/screens/AddEditClient/ContactInfo/MailingAddress.tsx

```diff
@@ -0,0 +1,126 @@
+import { Colors, Spacings } from '@monorepo/expo/shared/static';
+import {
+  BasicInput,
+  CardWrapper,
+  TextRegular,
+} from '@monorepo/expo/shared/ui-components';
+import { debounce } from '@monorepo/expo/shared/utils';
+import axios from 'axios';
+import { useCallback, useState } from 'react';
+import { useFormContext } from 'react-hook-form';
+import { TouchableOpacity, View } from 'react-native';
+import {
+  CreateClientProfileInput,
+  UpdateClientProfileInput,
+} from '../../../apollo';
+
+const apiUrl = process.env['EXPO_PUBLIC_API_URL'];
+
+export default function MailingAddress() {
+  const [suggestions, setSuggestions] = useState<any>([]);
+
+  const { setValue, watch } = useFormContext<
+    UpdateClientProfileInput | CreateClientProfileInput
+  >();
+
+  const mailingAddress = watch('mailingAddress') || '';
+
+  const onReset = () => {
+    setValue('mailingAddress', '');
+  };
+
+  const onSuggestionsSelect = (place: any) => {
+    setValue('mailingAddress', place.description.split(', ')[0]);
```

Comment:
this will give us city and state
```suggestion
    setValue('mailingAddress', place.description.split(', ').slice(0, 3).join(', ').toString());
```

Reply:
oh, wait. that doesn't get us zipcode. hmm

Reply:
according to gpt, the solution is kind of a bummer
> To include postal codes in the results returned by the Google Places API, you need to use the place_id from the autocomplete results to make a subsequent Place Details request. The autocomplete endpoint itself does not return detailed address components like postal codes.

Reply:
place.description.substring(
      0,
      place.description.lastIndexOf(',')
    );

    for safety did this way

Reply:
ready!

Reply:
nice! 🙏